src/devices/src/virtio/net/queue_handler.rs

```diff
@@ -0,0 +1,95 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR BSD-3-Clause
+
+use event_manager::{EventOps, Events, MutEventSubscriber};
+use log::error;
+use vm_memory::GuestAddressSpace;
+use vmm_sys_util::epoll::EventSet;
+use vmm_sys_util::eventfd::EventFd;
+
+use crate::virtio::SingleFdSignalQueue;
+
+use super::simple_handler::SimpleHandler;
+
+const TAPFD_DATA: u32 = 0;
+const RX_IOEVENT_DATA: u32 = 1;
+const TX_IOEVENT_DATA: u32 = 2;
+
+pub struct QueueHandler<M: GuestAddressSpace> {
+    pub inner: SimpleHandler<M, SingleFdSignalQueue>,
+    pub rx_ioevent: EventFd,
+    pub tx_ioevent: EventFd,
+}
+
+impl<M: GuestAddressSpace> QueueHandler<M> {
+    // Helper method that receives an error message to be logged and the `ops` handle
+    // which is used to unregister all events.
+    fn handle_error<S: AsRef<str>>(&self, s: S, ops: &mut EventOps) {
+        error!("{}", s.as_ref());
+        ops.remove(Events::empty(&self.rx_ioevent))
+            .expect("Failed to remove rx ioevent");
+        ops.remove(Events::empty(&self.tx_ioevent))
+            .expect("Failed to remove rx ioevent");
```

Comment:
```suggestion
            .expect("Failed to remove tx ioevent");
```

Reply:
I think it would be better to `ops.remove` them in a row, keep track of the results, and panic after all the `remove`s have been attempted so that events don't incorrectly remain in the event manager in case of a quick death.

That being said, I'm not sure about panicking in library code. There's not much we can do about it yet but if you go ahead with the `expect`s, please add an issue to track removing them once we have metrics.